plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.decoders;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.block.BlockBuilder;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class VarbinaryDecoder
+        implements Decoder
+{
+    @Override
+    public void decode(Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else {
+            checkState(value instanceof String, "Unexpected type for value '%s'", value.getClass());
+            Slice slice = Slices.wrappedBuffer(toBytes(value.toString()));
+            output.writeBytes(slice, 0, slice.length()).closeEntry();
+        }
+    }
+
+    private static byte[] toBytes(String stringValue)
```

Comment:
The same method exists in `PinotSegmentPageSource`. Please consider making this method public and reuse in `PinotSegmentPageSource`.